module/os/linux/zfs/zvol_os.c

```diff
@@ -953,6 +1229,11 @@ zvol_os_free(zvol_state_t *zv)
 	ASSERT0(zv->zv_open_count);
 	ASSERT3P(zv->zv_zso->zvo_disk->private_data, ==, NULL);
 
+#ifdef HAVE_BLK_MQ
+	if (zv->zv_zso->use_blk_mq)
+		flush_scheduled_work();
```

Comment:
Is this still needed since we're only using the taskq's now?  If so, can you add a comment explaining why.